pyproject.toml

```diff
@@ -54,6 +54,7 @@ dependencies = [
     "ssh2-python3==1.11.1",
     "tcconfig==0.29.1",
     "tenacity==9.0.0",
+    "pytest-subtests==0.14.2",
```

Comment:
Could we remove the usage of subtests in general? I dont like concept of subtests in pytest much and it would simplify a few functions. Also given the intrusivity of this PR I would not mind more test polish

Reply:
artifact tests are using it, and it fit nicely to their structure.

I don't see a reason to remove it.